File: TwoSum.py
#https://leetcode.com/problems/two-sum/



class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:

        j = len(nums)

        #O(n)
        nums = [(nums[i],i) for i in range(j)]

        #O(nlogn)
        numsSorted = sorted(nums,key=lambda x: x[0])

        i = 0
        j = len(nums) - 1


        #O(n)
        while i < j:
            val = numsSorted[i][0] + numsSorted[j][0]

            if val == target:
                return [numsSorted[i][1],numsSorted[j][1]]

            if val > target:
                j -= 1

            if val < target:
                i += 1

File: TwoSum.js
/* https://leetcode.com/problems/two-sum/description/ */

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const nums_checked = new Map();
    let i = 0;
    for (const x of nums) {
        let cur_dif = target - x;
        if (nums_checked.has(cur_dif)) {
            return[i, nums_checked.get(cur_dif)];
        }
        nums_checked.set(x,i);
        i++;
    }
    return [];
}